platform/openide.util/test/unit/src/org/openide/util/PairTest.java

```diff
@@ -19,39 +19,90 @@
 package org.openide.util;
 
 import org.netbeans.junit.NbTestCase;
-import org.openide.util.Pair;
 
 /**
  *
- * @author Tomas Zezula
+ * @author Tomas Zezula, Lukasz Bownik
  */
 public class PairTest extends NbTestCase {
 
-    public PairTest(String name) {
+    //--------------------------------------------------------------------------
+    public PairTest(final String name) {
+
         super(name);
     }
 
+    //-------------------------------------------------------------------------- 
+    public void test_pairEquality() {
+
+        Pair<Integer, Integer> pair11a = Pair.of(1, 1);
+        Pair<Integer, Integer> pair11b = Pair.of(1, 1);
+        Pair<Integer, Integer> pair12 = Pair.of(1, 2);
+        Pair<Integer, Integer> pairNul1 = Pair.of(null, 1);
+        Pair<Integer, Integer> pair1Null = Pair.of(1, null);
+        Pair<Integer, Integer> pairNullNull = Pair.of(null, null);
+
+        assertTrue(pair11a.equals(pair11a));
+        assertEquals(pair11a.hashCode(), pair11a.hashCode());
+        assertEquals(new Integer(1), pair11a.first());
+        assertEquals(new Integer(1), pair11a.second());
+
+        assertTrue(pair12.equals(pair12));
+        assertEquals(pair12.hashCode(), pair12.hashCode());
+        assertEquals(new Integer(1), pair12.first());
+        assertEquals(new Integer(2), pair12.second());
+
+        assertTrue(pair11a.equals(pair11b));
+        assertEquals(pair11a.hashCode(), pair11b.hashCode());
+        assertEquals(new Integer(1), pair11b.first());
+        assertEquals(new Integer(1), pair11b.second());
+
+        assertTrue(pair1Null.equals(pair1Null));
+        assertEquals(pair1Null.hashCode(), pair1Null.hashCode());
+        assertEquals(new Integer(1), pair1Null.first());
+        assertEquals(null, pair1Null.second());
+
+        assertTrue(pairNul1.equals(pairNul1));
+        assertEquals(pairNul1.hashCode(), pairNul1.hashCode());
+        assertEquals(null, pairNul1.first());
+        assertEquals(new Integer(1), pairNul1.second());
+
+        assertTrue(pairNullNull.equals(pairNullNull));
+        assertEquals(pairNullNull.hashCode(), pairNullNull.hashCode());
+        assertEquals(null, pairNullNull.first());
+        assertEquals(null, pairNullNull.second());
+    }
+
+    //-------------------------------------------------------------------------- 
+    public void test_pairInequality() {
```

Comment:
Please use camelCase and not snakeCase

Reply:
fixed in https://github.com/apache/netbeans/pull/5115/commits/68bc78dc9355a667f7d8208c9e0f39bf53c18013